api/controllers/OpenEAppController.js

```diff
@@ -96,7 +100,31 @@ const OpenEAppController = {
     },
 
     _formatDate(date) {
-        return moment(date).format('DD MMMM YYYY');
+        return dayjs(date).format('DD MMMM YYYY');
+    },
+
+    _calculateDaysLeftToDownload(req) {
+        const urlObj = new URL(
+            `${req.protocol}://${req.headers.host}${req.originalUrl}`
+        );
+        const queryParams = new URLSearchParams(urlObj.search);
+        if (!queryParams.has('completedDate')) {
+            throw new Error('No date value found in url');
+        }
+        const completedDate = queryParams.get('completedDate');
+        const currentDate = dayjs(Date.now());
+        const differenceBetweenCurrentAndCompletedDate = currentDate.diff(
+            completedDate,
+            'day'
+        );
+        const daysLeftToDownloadDoc =
+            MAX_DAYS_TO_LEGALISE -
+            Number(differenceBetweenCurrentAndCompletedDate);
+        const numberIsNegative = Math.sign(daysLeftToDownloadDoc) === -1;
+        if (numberIsNegative) {
+            throw new Error('Application has expired');
+        }
```

Comment:
maybe this?
```suggestion
        if (daysLeftToDownloadDoc < 0) {
            throw new Error('Application has expired');
        }
```

I appreciate verbosity when naming things, it makes things easier to read - but this might be too verbose or even adding complexity where there doesn't need to be. `< 0` is simpler (and less lines ;))

Reply:
fair point 😅

Reply:
You did submit the PR on a Friday 😄

Reply:
ha, that's true